docs: add SECURITY.md (#44)
charmed-hpc/slurmutils: SECURITY.md

# Security policy

## Supported versions

> Security updates will be released after vulnerabilities have been identified and fixed. At this point in time,
> we will only release security patches to the latest major version released on PyPI.

Reviewer: How would you feel about using this blurb instead for `slurmutils`?

```suggestion
Security updates will be released for all major versions that have had releases in the last year.
```

Basically says "any major version" within the last year. v0 is our major version currently, so we'll support it with security updates for the next year. Once we release v1 (stable API), we release security updates for v0 for the next year since not everyone will have switched over to v1 immediately. Little bit more brief, but shows we're committed to supporting the recent major releases.

_Got this from the `ops` team btw..._

Reviewer: > I think this is fine if breaking changes are infrequent, but it also means we would be committing to doing at most 1 breaking change per year, or we'll have to maintain 3+ versions at the same time...

Hmm... you bring up a good point here. I don't see us reaching v44 like `cryptography`, but yeah we could potentially work ourselves into a hole with supporting many releases at once, especially if we have cycle where we're all in on one of our libraries. Let's say the "latest 2 major versions" to narrow our scope, but then potentially expand as we start looking at LTS commitments.
